### DISCUSSION TITLE
Como criar um sistema de temas com varias cores

### DISCUSSION THREAD

**Discussion by andre-mendes** [frontendbr/forum#385] [Perguntas e Respostas]
Fala rapaziada, tanquilo? :D 

Aqui na empresa estamos montando um projeto onde o cliente vai ter a opção de escolher mais de um tema para seu layout.... e pra cada layout, ele também vai poder escolher as cores do tema.

A dúvida é o seguinte...

Qual a melhor forma de fazer isso? Eu teria que ter um arquivo pro CSS base e um arquivo pras cores?

E se eu tiver la 20 cores diferente, terei que ter 20 arquivos só pra cor?

E se for 1 arquivo com tudo, e precisar alterar algo no layout, terei que compilar 20x pra ter os 20 temas das cores?

Variáveis nativas ja estão massa pra se usar? O público desse site usa mto navegador velho :P 

O que vc sugerem? se ficou alguma dúvida, comenta ae que eu tenho explicar melhor 

vlw \o/

**Comment by batoreh** [comment]
Se eu fosse você, separaria as cores alteráveis do layout em um arquivo só utilizando variáveis de algum preprocessador e então automatizaria a alteração dessas cores com algum script (acho que em python ou node será o mais simples, caso vc não tenha experiência com isso). O script basicamente pegaria inputs do usuário e atribuiria às variáveis no arquivo.

Eu só não entendi muito bem como vocês pretendem captar essas opções dos clientes. Se for via interface, com `<input type="color">`, fica bem tranquilo pra pegar o valor dessas cores, mas também vai depender da arquitetura de vocês, do que você pode ou não fazer, de como será o processo de build desses temas e de como será o contato com o cliente.

**Comment by RicardoJunior** [comment]
Cara, da uma olhada nisso:

http://codepen.io/rmjunior/pen/BKdGbe

Eu fiz alguma solução por essa linha. Com uma idéia de palhetas de cores.
